File: source/math/Vertex.java
package math;

import java.io.Serializable;

public class Vertex implements Serializable {
    public Float3 world;
    public Float2 texture;
    public Float3 normal;

    public Vertex() {
        world = new Float3();
        texture = new Float2();
        normal = new Float3();
    }

    public Vertex(Float3 world) {
        this.world = world;
        texture = new Float2();
        normal = new Float3();
    }

    public Vertex(Float3 world, Float2 texture) {
        this.world = world;
        this.texture = texture;
        normal = new Float3();
    }

    public Vertex(Float3 world, Float2 texture, Float3 normal) {
        this.world = world;
        this.texture = texture;
        this.normal = normal;
    }

    public Vertex(Vertex vertex) {
        world = new Float3(vertex.world);
        texture = new Float2(vertex.texture);
        normal = new Float3(vertex.normal);
    }

    public String toString() {
        return "(" + world + ", " + texture + ", " + normal + ")";
    }
}

File: source/math/Vertex.kt
package math

import java.io.Serializable

class Vertex : Serializable {
    var world: Float3?
    var texture: Float2?
    var normal: Float3?

    constructor() {
        world = Float3()
        texture = Float2()
        normal = Float3()
    }

    constructor(world: Float3?) {
        this.world = world
        texture = Float2()
        normal = Float3()
    }

    constructor(world: Float3?, texture: Float2?) {
        this.world = world
        this.texture = texture
        normal = Float3()
    }

    constructor(world: Float3?, texture: Float2?, normal: Float3?) {
        this.world = world
        this.texture = texture
        this.normal = normal
    }

    constructor(vertex: Vertex) {
        world = Float3(vertex.world)
        texture = Float2(vertex.texture)
        normal = Float3(vertex.normal)
    }

    override fun toString(): String {
        return "($world, $texture, $normal)"
    }
}